### PR TITLE
Step2

### PR DESCRIPTION
**Pull request by eshaham**
Create a movie details page and navigate to it from the list of movies.
The movie class was made `Parcelable` and is being passed to the movie details activity.

image

### CODE REVIEW

**Comment by ifeins** [comment]
Maybe some visual adjustments?
Maybe placing the poster at the top (possibly at the center) and have the description and rating below it?

File: app/src/main/java/com/eshaham/moviefiend/activities/MovieDetailsActivity.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    private void initMovieView(Movie movie) {
+        setTitle(movie.title);
+
+        TextView descriptionView = (TextView) findViewById(R.id.description_text_view);
+        descriptionView.setText(movie.description);
+
+        ImageView imageView = (ImageView) findViewById(R.id.poster_image_view);
+        String basePosterUrl = getString(R.string.movie_db_base_poster_url);
+        Picasso.with(this)
+                .load(basePosterUrl + movie.posterPath)
```
Comment (ifeins) [inline app/src/main/java/com/eshaham/moviefiend/activities/MovieDetailsActivity.java]: `Uri.Builder`

File: app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java [hunk 2]
```diff
@@ -36,42 +39,61 @@ public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
         private final Context mContext;
-        private final String mBasePosterUrl;
+        private final Uri mBasePosterUri;
 
         private final ImageView mImageView;
         private final TextView mTitleTextView;
         private final TextView mRatingTextView;
 
-        public ViewHolder(Context mContext, View view, String mBasePosterUrl) {
+        public ViewHolder(Context mContext, View view, Uri mBasePosterUri) {
```
Comment (ifeins) [inline app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java]: method parameters should not use the `m` notation

File: app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java [hunk 3]
```diff
@@ -36,42 +39,61 @@ public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
             super(view);
 
             this.mContext = mContext;
-            this.mBasePosterUrl = mBasePosterUrl;
+            this.mBasePosterUri = mBasePosterUri;
```
Comment (ifeins) [inline app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java]: no need for `this`

File: app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java [hunk 4]
```diff
@@ -36,42 +39,61 @@ public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
                     .fit()
                     .into(mImageView);
             mTitleTextView.setText(movie.title);
             mRatingTextView.setText(String.valueOf(movie.rating));
         }
     }
+
+    public interface ClickListener {
+        void onMovieClicked(Movie movie);
+    }
```
Comment (ifeins) [inline app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java]: This should be at the top of the class: https://github.paypal.com/cpc/consumer-venice-android/blob/project_style_guide/Documentation/project_style_guidelines.md#2216-class-member-ordering

File: app/src/main/java/com/eshaham/moviefiend/model/Movie.java [hunk 5]
```diff
@@ -1,16 +1,57 @@
...
+    @SerializedName("overview")
+    public String description;
+
     @SerializedName("poster_path")
     public String posterPath;
 
     @SerializedName("vote_average")
     public double rating;
+
+    public Movie(Parcel parcel) {
```
Comment (ifeins) [inline app/src/main/java/com/eshaham/moviefiend/model/Movie.java]: For the `Parcelable` implementation you need to add an editor-fold

**Comment by eshaham** [comment]
@ifeins updated PR with screenshot of the modified movie details view

File: app/src/main/java/com/eshaham/moviefiend/activities/MovieDetailsActivity.java (hunk 1 above)
Comment (eshaham) [inline app/src/main/java/com/eshaham/moviefiend/activities/MovieDetailsActivity.java]: @ifeins deda2c4313ca5fae62cc62818a0a76f56ee00364

File: app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java (hunk 2 above)
Comment (eshaham) [inline app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java]: @ifeins oops 6d0172d

File: app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java (hunk 3 above)
Comment (eshaham) [inline app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java]: @ifeins 6d0172d

File: app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java (hunk 4 above)
Comment (eshaham) [inline app/src/main/java/com/eshaham/moviefiend/adapters/MovieListAdapter.java]: @ifeins 6cb3725

File: app/src/main/java/com/eshaham/moviefiend/model/Movie.java (hunk 5 above)
Comment (eshaham) [inline app/src/main/java/com/eshaham/moviefiend/model/Movie.java]: @ifeins 682951c

**Comment by ifeins** [comment]
Woah, looks nice, I like the header 👍

File: app/src/main/java/com/eshaham/moviefiend/model/Movie.java [hunk 6]
```diff
@@ -1,16 +1,66 @@
...
+                }
+
+                public Movie[] newArray(int size) {
+                    return new Movie[size];
+                }
+            };
+
+    // </editor-fold desc="Parcelable">
+
+    private void readFromParcel(Parcel parcel) {
```
Comment (ifeins) [inline app/src/main/java/com/eshaham/moviefiend/model/Movie.java]: Not that it's critical, but what are you gaining by extracting this to a separate method and not putting it inline in the constructor? It's not going to be used anywhere else

File: app/src/main/res/layout/activity_movie_details.xml [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                <RatingBar
+                    android:id="@+id/rating_bar"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="5dp"
+                    android:layout_marginBottom="5dp"
+                    android:isIndicator="true"
+                    android:numStars="5"
+                    tools:rating="4.5"
+                    style="?android:attr/ratingBarStyleIndicator"/>
```
Comment (ifeins) [inline app/src/main/res/layout/activity_movie_details.xml]: According to our conventions the `style` attribute should be at the top (below `android:id`)
This enables you to inherit from a style but still override some of the style properties.
If you put the style at the bottom then any properties you declared above will not be overridden (I think).

File: app/src/main/res/layout/activity_movie_details.xml [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+                android:layout_height="wrap_content"
+                android:adjustViewBounds="true"
+                android:layout_marginRight="10dp"
+                android:layout_marginEnd="10dp"
+                android:layout_gravity="center_horizontal"
+                tools:src="@drawable/movie"/>
+
+        </LinearLayout>
+
+
```
Comment (ifeins) [inline app/src/main/res/layout/activity_movie_details.xml]: Some excess newlines, you can run `Cmd + Option + L` to format XML files as well.

File: app/src/main/java/com/eshaham/moviefiend/model/Movie.java (hunk 6 above)
Comment (eshaham) [inline app/src/main/java/com/eshaham/moviefiend/model/Movie.java]: @ifeins usually prefer not to have too many lines of code in my constructor. in this case it does feel redundant, so aa7353a

File: app/src/main/res/layout/activity_movie_details.xml (hunk 7 above)
Comment (eshaham) [inline app/src/main/res/layout/activity_movie_details.xml]: d0a627b

File: app/src/main/res/layout/activity_movie_details.xml (hunk 8 above)
Comment (eshaham) [inline app/src/main/res/layout/activity_movie_details.xml]: @ifeins fdf6404

**Pull request merged**